### ISSUE TITLE
Inline JSX elements (CamelCase) do not seem syntax highlighted

### ISSUE THREAD

**Issue by balupton** [silvenon/vscode-mdx#9]
They are treated just like markdown. Lowercasing the jsx element works for syntax highlighting, but then it doesn't function.

**Comment by silvenon** [comment]
Could you share a code snippet and/or screenshot? I don't understand what isn't syntax highlighted.

**Comment by silvenon** [comment]
Ah, I think you're referring to **inline** JSX, correct?

**Comment by balupton** [comment]
Correct. Sorry, posted the issue on my phone. An example is a JSX element (so CamelCase) within a nested markdown list

**Comment by balupton** [comment]
<img width="650" alt="screen shot 2019-01-29 at 4 31 32 pm" src="https://user-images.githubusercontent.com/61148/51886298-5a3ce780-23e3-11e9-91fb-05f60bec5cb3.png">
<img width="268" alt="screen shot 2019-01-29 at 4 31 56 pm" src="https://user-images.githubusercontent.com/61148/51886308-6628a980-23e3-11e9-9da1-3cf3efdc1684.png">

**Comment by silvenon** [comment]
Yeah, in MDX that's what's classified as inline JSX, where a single line needs to be partially interpreted as Markdown and partially as JSX.

I'll see what I can do, thanks for reporting. 🙂

**Comment by blummis** [comment]
Hi, any ideas how to fix this?

This is how it shows for me:

<img width="259" alt="inline-tags" src="https://user-images.githubusercontent.com/101323/61171848-ddaf5700-a585-11e9-8f6d-9b87df0675fc.png">

**Comment by silvenon** [comment]
In the near future no, I'm sorry, I'm very inexperienced with grammars in general. I think I need to rebuild this extension entirely to gain more more power over these cases because it's currently only looking for empty lines, as far as I can remember, but MDX is more complex now, with inline JSX, interpolations etc.

**Comment by karlhorky** [comment]
Because HTML tags are also valid when written uppercase, and web components would be classified as "Unknown", maybe this can be fixed in the base `text.html.markdown` grammar, which also has this issue:

![Screen Shot 2019-08-14 at 15 57 10](https://user-images.githubusercontent.com/1935696/63026993-6e8e9080-beac-11e9-8605-47b9aea6902e.png)

Compare that with the HTML grammar highlighting:

![Screen Shot 2019-08-14 at 16 02 35](https://user-images.githubusercontent.com/1935696/63027367-202dc180-bead-11e9-9ab9-2693f7385b48.png)

I am also not experienced with grammars, but from a cursory look it seems like you're using here:

https://github.com/silvenon/vscode-mdx/blob/fefe05634acc491a0b7c158840d7bb409a5d9605/syntaxes/mdx.tmLanguage.json#L41-L46

I will open an issue on the grammar to request feedback on this change.

**Comment by karlhorky** [comment]
Opened https://github.com/microsoft/vscode/issues/79119 for this.

**Edit:** Copied to microsoft/vscode-markdown-tm-grammar#49, in case that is the correct place for it.

**Comment by karlhorky** [comment]
Caveat: JSX highlighting would be of course better if someone can crack it (cc @mjbvz, could you possibly point us in the right direction for updating [the grammar](https://github.com/silvenon/vscode-mdx/blob/master/syntaxes/mdx.tmLanguage.json)?) - because it would show a different color for capitalized tags, but this would at least be an improvement on the current situation.

**Comment by karlhorky** [comment]
Well case insensitivity was added in https://github.com/microsoft/vscode-markdown-tm-grammar/commit/67b9b347e34950f3e63ec923e20ac620ffa2ddd7 - should be out in the next VS Code Insiders build.

But unknown / custom elements were not addressed. I've asked about that in https://github.com/microsoft/vscode-markdown-tm-grammar/issues/49#issuecomment-521658427

**Comment by karlhorky** [comment]
Unknown / custom elements merged in! https://github.com/microsoft/vscode-markdown-tm-grammar/pull/54

Still is not a complete fix, because other JSX syntax like non-string props passing (`<Grid style={{ background: "green" }}>`) is still not highlighted, but it's a start for simple components!

**Comment by silvenon** [comment]
Great! I'll see what I can do about that. 👍 Thanks for keeping this issue updated!

**Comment by karlhorky** [comment]
No problem :) yeah I think the "correct" way to fix it would be for inline tags (probably all inline tags) to start triggering the JSX grammar again. Not entirely sure how to do this off the top of my head, but I guess the `vscode-markdown-tm-grammar` code could be illuminating in this regard... They switch to the other `html` grammar a lot within it.

**Comment by karlhorky** [comment]
Seems like looking at the Markdown grammar will be helpful in other ways too. For example, currently in the MDX grammar, the Markdown inside the HTML element is not highlighted (Markdown can be embedded within HTML if it starts at the start of the line):

![Screen Shot 2019-08-28 at 12 20 38](https://user-images.githubusercontent.com/1935696/63847646-7ceda980-c98e-11e9-8dd4-0a8dd37506af.png)

Compare that to the Markdown grammar:

![Screen Shot 2019-08-28 at 12 20 28](https://user-images.githubusercontent.com/1935696/63847602-69dad980-c98e-11e9-9e9a-4836ec2fae1c.png)

**Comment by karlhorky** [comment]
The fix I refer to in https://github.com/silvenon/vscode-mdx/issues/9#issuecomment-522855414 above was released in VS Code 1.38 btw:

https://code.visualstudio.com/updates/v1_38#_notable-fixes

> [79478](https://github.com/microsoft/vscode/issues/79478): Highlight custom html tags in markdown files

**Comment by felixakiragreen** [comment]
This was fixed in the VSCode Markdown lib but still broken in this one.
